tests/snmp/airos.py

```diff
@@ -1,27 +1,48 @@
 import unittest
 from netengine.backends.snmp import AirOS
 from netengine.exceptions import NetEngineError
+from pysnmp.entity.rfc3413.oneliner import cmdgen
 
 from ..settings import settings
+from ..static import MockOutputMixin, SpyMock
 
 
 __all__ = ['TestSNMPAirOS']
 
 
-class TestSNMPAirOS(unittest.TestCase):
+class TestSNMPAirOS(unittest.TestCase, MockOutputMixin):
     
     def setUp(self):
         self.host = settings['airos-snmp']['host']
         self.community = settings['airos-snmp']['community']
         self.port = settings['airos-snmp'].get('port', 161)
-        
         self.device = AirOS(self.host, self.community, port=self.port)
+
+        # mock calls being made to devices
+        self.oid_mock_data = self._load_mock_json('/test-airos-snmp.json')
+        self.nextcmd_patcher = SpyMock._patch(
+            target=cmdgen.CommandGenerator,
+            attribute='nextCmd',
+            wrap_obj=self.device._command,
+            return_value=[0, 0, 0, [[[0, 1]]] * 5]
+        )
+        self.getcmd_patcher = SpyMock._patch(
+            target=cmdgen.CommandGenerator,
+            attribute='getCmd',
+            wrap_obj=self.device._command,
+            side_effect=lambda *args: self._get_mocked_getcmd(
+                data=self.oid_mock_data, input=args
+            ),
+        )
+        self.getcmd_patcher.start()
     
     def test_get_value_error(self):
+        self.getcmd_patcher.stop()
         with self.assertRaises(NetEngineError):
             self.device.get_value('.')
```

Comment:
Following up to my previous comment, here we don't know that exactly raised `NetEngineError`. Was it raised because connection to the device was not established? Or, was it raised because `.` can't be resolved to a node?